src/bc_protocol/connection.rs

```diff
@@ -115,6 +129,14 @@ impl BcConnection {
 
         Ok(())
     }
+
+    pub fn set_encrypted(&self, value: EncryptionProtocol) {
+        *(self.encryption_protocol.lock().unwrap()) = value;
```

Comment:
Nice. This is where I got stuck, I was trying to apply the encryption setup before starting the 

It's still possible at the data structure level for different receivers to race and change encryption on the fly, but in practice this will never happen because login is the first thing that happens.

Reply:
Yeah two sticking point on the model were this and having to share the encycption data in both ser and der.